soc/espressif/esp32c3/default.ld

```diff
@@ -793,14 +802,6 @@ SECTIONS
   .flash.rodata_end : ALIGN(0x10)
   {
     . = ALIGN(4);
-
```

Comment:
Is this related to this change? It would not deserve a proprr commit that explains why it is being removed, since the relationship to XIP seems to be not that explicit?

Reply:
It is indeed related to XIP. Should stay here.
Background: ' __tdata_start' definition depends whether `XIP` is or no set. Current snippet was a workaround to fix TLs stuff.